src/core/client/admin/components/MediaContainer/MediaContainer.tsx

```diff
@@ -20,46 +17,18 @@ const MediaContainer: FunctionComponent<Props> = ({ comment }) => {
     return null;
   }
 
-  switch (media.__typename) {
-    case "GiphyMedia":
-      return (
-        <GiphyMedia
-          still={media.still}
-          video={media.video}
-          title={media.title}
-          width={media.width}
-          height={media.height}
-        />
-      );
-    case "ExternalMedia":
-      return (
-        <ExternalMedia
-          id={comment.id}
-          url={media.url}
-          siteID={comment.site.id}
-        />
-      );
-    case "TwitterMedia":
-      return (
-        <TwitterMedia
-          id={comment.id}
-          url={media.url}
-          siteID={comment.site.id}
-        />
-      );
-    case "YouTubeMedia":
-      return (
-        <YouTubeMedia
-          id={comment.id}
-          url={media.url}
-          siteID={comment.site.id}
-          still={media.still}
-          title={media.title}
-        />
-      );
-    case "%other":
-      return null;
+  if (media.__typename === "%other") {
+    return null;
   }
+
+  return (
+    <Media
+      media={media}
+      type={media.__typename}
+      id={comment.id}
+      siteID={comment.site.id}
+    />
+  );
```

Comment:
The disadvantage to doing this instead of what was here before is that you can't rely on the type system enough, the connection is a lot more "loose".

Passing the type name information into the `<Media />` component relies on that component to understand the prop requirements and shape of the items based on the passed `__typename`.